cosmos/x/evm/plugins/txpool/mempool/mempool.go

```diff
@@ -70,6 +71,13 @@ func (etp *EthTxPool) Insert(ctx context.Context, tx sdk.Tx) error {
 	etp.mu.Lock()
 	defer etp.mu.Unlock()
 
+	// Reject txs with a nonce lower than the nonce reported by the statedb.
+	if sdbNonce := etp.nr.GetNonce(
+		common.BytesToAddress(tx.GetMsgs()[0].GetSigners()[0]),
+	); sdbNonce > evmtypes.GetAsEthTx(tx).Nonce() {
+		return errors.New("nonce too low")
+	}
```

Comment:
if you send a tx with a nonce lower than pending nonce, should it also fail or just replace the existing tx?

Reply:
This is the StateDB nonce, so it is only the highest that has been processed. 

We need to add logic for replacements yeah u right

Reply:
to replace an existing tx `tx1` w a new tx `tx2` that has the same nonce:
- max base fee per gas must be 10% more on `tx2` than on `tx1`
- max tip per gas must be 10% more on `tx2` than on `tx1`

source: https://www.youtube.com/watch?v=c4N79UXZqSc&ab_channel=ETHGlobal

Reply:
jokes on me: https://github.com/berachain/polaris/pull/739/commits/928cf1b92d10fe7181645a30f22c5587b7bb6bae